### PR TITLE
Linted the index.js file and edited eslint config a bit

### PR DESCRIPTION
**Pull request by QuantumlyTangled**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -33,8 +33,11 @@ module.exports = class Clash {
 
     async getPlayer(tag) {
         if (!tag) throw new TypeError('Invalid arguments: this function expects a player tag.');
-        if (typeof tag !== 'string') throw new TypeError('Clan tag must be a string.');
-        const _tag = encodeURIComponent(tag.toString()).toUpperCase();
+        if (typeof tag !== 'string') throw new TypeError('Player tag must be a string!');
+        const _tag_ = await tag.toUpperCase;
```
Comment (Soumil07) [inline src/index.js]: awaiting String#toUpperCase?

Comment (QuantumlyTangled) [reply src/index.js]: Did the requested change

File: src/index.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ module.exports = class Clash {
     async getPlayer(tag) {
         if (!tag) throw new TypeError('Invalid arguments: this function expects a player tag.');
         if (typeof tag !== 'string') throw new TypeError('Player tag must be a string!');
-        const _tag_ = await tag.toUpperCase;
+        const _tag_ = tag.toUpperCase;
```
Comment (Soumil07) [inline src/index.js]: can be clubbed into ```javascript
const _tag = /[#][0289PYLQGRJCUV]{4,9}/i.test(tag.toUpperCase());
```

File: src/index.js [hunk 3]
```diff
@@ -70,7 +70,7 @@ module.exports = class Clash {
     async getClan(tag) {
         if (!tag) throw new TypeError('Invalid arguments: this function expects a clan tag.');
         if (typeof tag !== 'string') throw new TypeError('Clan tag must be a string.');
-        const _tag_ = await tag.toUpperCase;
+        const _tag_ = tag.toUpperCase;
```
Comment (Soumil07) [inline src/index.js]: same here

File: src/index.js [hunk 4]
```diff
@@ -96,7 +96,7 @@ module.exports = class Clash {
     async getWarLog(tag) {
         if (!tag) throw new TypeError('Invalid arguments: this function expects a clan tag.');
         if (typeof tag !== 'string') throw new TypeError('Clan tag must be a string.');
-        const _tag_ = await tag.toUpperCase;
+        const _tag_ = tag.toUpperCase;
```
Comment (Soumil07) [inline src/index.js]: and here

**Pull request merged**